libs/blocks/marketo/marketo.js

```diff
@@ -186,11 +188,73 @@ export const formSuccess = (formEl, formData) => {
   return false;
 };
 
+export const handleIframeTimeout = (el) => {
+  const searchParams = new URLSearchParams(window.location.search);
+  const config = getConfig();
+  const iframe = document.querySelector('iframe[src*="/index.php/form/XDFrame"]');
+  const formEl = el.querySelector('form');
+  let iframeTimeout = null;
+
+  const handleIframeReady = (event) => {
+    if (event.origin === 'https://engage.adobe.com') {
+      const message = JSON.parse(event.data);
+      if (message.mktoReady) {
+        if (iframeTimeout) clearTimeout(iframeTimeout);
+        const errorOverlay = el.querySelector('.marketo-overlay');
+        if (errorOverlay) {
+          errorOverlay.remove();
+          if (formEl) formEl.inert = false;
+        }
+        window.removeEventListener('message', handleIframeReady);
+      }
+    }
+  };
+
+  const decorateOverlay = async () => {
+    const cookieSize = document.cookie.length > 4096 ? '> 4k' : '< 4k';
+    window.lana?.log(`Marketo iframe timeout - Cookie Size ${cookieSize}`, { tags: 'error,marketo' });
+
+    if (el.querySelector('.marketo-overlay')) return;
+    if (!config.marketo?.showError && searchParams.get('marketoOverlay') !== 'error') return;
+
+    const marketoErrorText = await replaceKey('marketo-load-error', config);
+    if (formEl) formEl.inert = true;
+
+    const errorMessage = createTag('p', { class: 'error' }, marketoErrorText);
+    const retryButton = createTag('button', { class: 'retry-button' }, 'Try Again');
```

Comment:
Does this need to be placeholder text so it can be translated?

Reply:
Good call